packages/fast-components-react-base/src/slider/slider.tsx

```diff
@@ -1135,10 +1135,14 @@ class Slider extends Foundation<SliderHandledProps, SliderUnhandledProps, Slider
             return value;
         }
 
-        return (
-            Math.floor((value - this.props.range.minValue) / step) * step +
-            this.props.range.minValue
-        );
+        let constrainedValue: number =
+            Math.floor((value - this.props.range.minValue) / step) * step;
+        constrainedValue =
+            (value - constrainedValue) / step >= 0.5
+                ? constrainedValue + step
+                : constrainedValue;
+
+        return constrainedValue + this.props.range.minValue;
```

Comment:
Can you not use `%`:
```ts
const remainder = value % step;
const updatedValue = remainder >= (step / 2) // check to see if this is over half a single step
    ? value - remainder + step // if so add a step
    : value - remainder;

return updatedValue <= this.props.range.minValue
    ? this.props.range.minValue
    : updatedValue > this.props.range.maxValue
        ? this.props.range.maxValue
        : updatedValue;
```

In this way you can skip your if (step === 0) as well.

Reply:
Range checking is a different fn, so:
       const remainder = value % step;
        const constrainedValue = remainder >= (step / 2) // check to see if this is over half a single step
            ? value - remainder + step // if so add a step
            : value - remainder;

        return constrainedValue + this.props.range.minValue;